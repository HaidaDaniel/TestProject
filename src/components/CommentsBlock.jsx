/** @format */
import PropTypes from 'prop-types'
import { Stack } from 'react-bootstrap'

import { Comment } from './index'
import { commentShape } from './PropTypes/DetailedProductShape'
import { StyledCommentsBlock } from '../styled/CommentsBlockStyles'

const CommentsBlock = ({ comments }) => (
  <StyledCommentsBlock>
    <div className='comment-block mt-2'>
      <h5>Customer Reviews :</h5>
      <Stack gap={3}>
        {comments &&
          comments.map((comment) => (
            <Comment
              key={comment.commentId}
              author={comment.author}
              text={comment.text}
              rating={comment.rating}
            />
          ))}
      </Stack>
    </div>
  </StyledCommentsBlock>
);

export default CommentsBlock

CommentsBlock.propTypes = {
  comments: PropTypes.arrayOf(PropTypes.shape(commentShape)).isRequired,
}
