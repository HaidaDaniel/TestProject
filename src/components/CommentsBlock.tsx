/** @format */
import { FC } from 'react'
import { Stack } from 'react-bootstrap'

import { Comment } from './'

import { StyledCommentsBlock } from '../styled/CommentsBlockStyles'
import { IComment } from './interfaces/'

interface CommentsBlockProps {
  comments: IComment[]
}

const CommentsBlock: FC<CommentsBlockProps> = ({ comments }) => (
  <StyledCommentsBlock>
    <div className='comment-block mt-2'>
      <h5>Customer Reviews :</h5>
      <Stack gap={3}>
        {comments &&
          comments.map((comment: IComment) => (
            <Comment
              key={comment.commentId}
              author={comment.author}
              text={comment.text}
              rating={comment.rating}
              commentId={comment.commentId}
            />
          ))}
      </Stack>
    </div>
  </StyledCommentsBlock>
)

export default CommentsBlock
