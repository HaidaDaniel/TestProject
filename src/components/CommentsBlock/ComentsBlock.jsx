/** @format */
import Comment from '../Comment/Comment'
import { Stack } from 'react-bootstrap'

function CommentsBlock({ comments }) {
  return (
    <div className='comment-block'>
      <h4>Customer Reviews :</h4>
      <Stack gap={3}>
        {comments &&
          comments.map((comment) => (
            <Comment
              key={comment.commentId}
              author={comment.author}
              text={comment.text}
              rating={comment.rating}
            />
          ))}
      </Stack>
    </div>
  )
}
export default CommentsBlock
