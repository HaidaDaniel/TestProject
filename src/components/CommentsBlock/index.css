.comment-block {
    margin-top: 1rem;
}