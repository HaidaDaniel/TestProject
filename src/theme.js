const theme = {
    colors: {
        primary: '#0d6efd',
        secondary: '#6c757d',
        success: '#28a745',
        error: '#dc3545',
        golden: '#f5a623'
    },
    fonts: {
        body: 'Arial, sans-serif',
        heading: 'Helvetica, sans-serif',
    },

};

export default theme;