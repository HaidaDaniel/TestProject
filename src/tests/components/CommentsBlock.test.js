import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import '@testing-library/jest-dom/extend-expect';

<<<<<<< HEAD
<<<<<<< HEAD
import CommentsBlock from '../../components/CommentsBlock';

import theme from '../../theme';
=======
import theme from '../../theme';
import CommentsBlock from '../../components/CommentsBlock';
>>>>>>> origin/newTest
=======
import theme from '../../theme';
import CommentsBlock from '../../components/CommentsBlock';
>>>>>>> cdcb47e (1st refactor of tests description)

const mockedComments = [
    {
        commentId: 1,
        author: 'John Doe',
        text: 'Great product!',
        rating: 5,
    },
    {
        commentId: 2,
        author: 'Jane Smith',
        text: 'Could be better.',
        rating: 3,
    },
];

<<<<<<< HEAD
<<<<<<< HEAD
test('should render CommentsBlock component with comments', () => {
=======
test('should renders CommentsBlock component with comments', () => {
>>>>>>> origin/newTest
=======
test('should renders CommentsBlock component with comments', () => {
>>>>>>> cdcb47e (1st refactor of tests description)
    render(<ThemeProvider theme={theme}><CommentsBlock comments={mockedComments} /></ThemeProvider>);

    const commentBlockElement = screen.getByTestId('comments-block');
    const commentElements = screen.getAllByTestId('comment');

    expect(commentBlockElement).toBeInTheDocument();
    expect(commentElements.length).toBe(mockedComments.length);

    mockedComments.forEach((comment, index) => {
        const commentElement = commentElements[index];
        expect(commentElement).toHaveTextContent(comment.text);
        expect(commentElement).toHaveTextContent(comment.author);
        expect(commentElement).toHaveTextContent(`★★★★★${comment.rating} /5`);
        expect(document.body).toMatchSnapshot();
    });
});

test('should render CommentsBlock component with no comments', () => {
    render(<ThemeProvider theme={theme}><CommentsBlock comments={[]} /></ThemeProvider>);

    const commentBlockElement = screen.getByTestId('comments-block');
    const noCommentsMessage = screen.getByText('No review available.Be the first reviewer!');

    expect(commentBlockElement).toBeInTheDocument();
    expect(noCommentsMessage).toBeInTheDocument();
    expect(document.body).toMatchSnapshot();
});
