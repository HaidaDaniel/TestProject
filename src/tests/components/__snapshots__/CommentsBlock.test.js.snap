// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CommentsBlock component with no comments 1`] = `
<body>
  <div>
    <div
      class="sc-bZCZoC qjAAu"
    >
      <div
        class="comment-block mt-2"
        data-testid="comments-block"
      >
        <h5>
          Customer Reviews :
        </h5>
        <div
          class="vstack gap-3"
        >
          <div>
            No review available.Be the first reviewer!
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`should renders CommentsBlock component with comments 1`] = `
<body>
  <div>
    <div
      class="sc-bZCZoC qjAAu"
    >
      <div
        class="comment-block mt-2"
        data-testid="comments-block"
      >
        <h5>
          Customer Reviews :
        </h5>
        <div
          class="vstack gap-3"
        >
          <div
            data-testid="comment"
          >
            <div
              class="sc-ksJisA hKuXOJ card"
            >
              <div
                class="sc-fnOeiS dGYhpH card-header"
              >
                <div
                  class="sc-jIJgYh gPIvtQ"
                >
                  <div
                    class="sc-jdkVqZ ivvINy"
                  >
                    <span
                      class="sc-hhWzdI sc-gXCJSa gPjntB fqfOft"
                      data-testid="filled"
                    >
                      ★
                    </span>
                  </div>
                  <div
                    class="sc-jdkVqZ ivvINy"
                  >
                    <span
                      class="sc-hhWzdI sc-gXCJSa gPjntB fqfOft"
                      data-testid="filled"
                    >
                      ★
                    </span>
                  </div>
                  <div
                    class="sc-jdkVqZ ivvINy"
                  >
                    <span
                      class="sc-hhWzdI sc-gXCJSa gPjntB fqfOft"
                      data-testid="filled"
                    >
                      ★
                    </span>
                  </div>
                  <div
                    class="sc-jdkVqZ ivvINy"
                  >
                    <span
                      class="sc-hhWzdI sc-gXCJSa gPjntB fqfOft"
                      data-testid="filled"
                    >
                      ★
                    </span>
                  </div>
                  <div
                    class="sc-jdkVqZ ivvINy"
                  >
                    <span
                      class="sc-hhWzdI sc-gXCJSa gPjntB fqfOft"
                      data-testid="filled"
                    >
                      ★
                    </span>
                  </div>
                  5
                   
                  /5
                </div>
              </div>
              <div
                class="sc-hBpgZr fzdmTd card-body"
              >
                <p
                  class="sc-fEyylQ bjhgma card-text"
                >
                  Great product!
                </p>
                <p
                  class="sc-bjEwCx bPzSwy card-text"
                >
                  author:
                  John Doe
                   
                  <br />
                   review id:
                  1
                </p>
              </div>
            </div>
          </div>
          <div
            data-testid="comment"
          >
            <div
              class="sc-ksJisA hKuXOJ card"
            >
              <div
                class="sc-fnOeiS dGYhpH card-header"
              >
                <div
                  class="sc-jIJgYh gPIvtQ"
                >
                  <div
                    class="sc-jdkVqZ ivvINy"
                  >
                    <span
                      class="sc-hhWzdI sc-gXCJSa gPjntB fqfOft"
                      data-testid="filled"
                    >
                      ★
                    </span>
                  </div>
                  <div
                    class="sc-jdkVqZ ivvINy"
                  >
                    <span
                      class="sc-hhWzdI sc-gXCJSa gPjntB fqfOft"
                      data-testid="filled"
                    >
                      ★
                    </span>
                  </div>
                  <div
                    class="sc-jdkVqZ ivvINy"
                  >
                    <span
                      class="sc-hhWzdI sc-gXCJSa gPjntB fqfOft"
                      data-testid="filled"
                    >
                      ★
                    </span>
                  </div>
                  <div
                    class="sc-jdkVqZ ivvINy"
                  >
                    <span
                      class="sc-hhWzdI gPjntB"
                      data-testid="empty"
                    >
                      ★
                    </span>
                  </div>
                  <div
                    class="sc-jdkVqZ ivvINy"
                  >
                    <span
                      class="sc-hhWzdI gPjntB"
                      data-testid="empty"
                    >
                      ★
                    </span>
                  </div>
                  3
                   
                  /5
                </div>
              </div>
              <div
                class="sc-hBpgZr fzdmTd card-body"
              >
                <p
                  class="sc-fEyylQ bjhgma card-text"
                >
                  Could be better.
                </p>
                <p
                  class="sc-bjEwCx bPzSwy card-text"
                >
                  author:
                  Jane Smith
                   
                  <br />
                   review id:
                  2
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`should renders CommentsBlock component with comments 2`] = `
<body>
  <div>
    <div
      class="sc-bZCZoC qjAAu"
    >
      <div
        class="comment-block mt-2"
        data-testid="comments-block"
      >
        <h5>
          Customer Reviews :
        </h5>
        <div
          class="vstack gap-3"
        >
          <div
            data-testid="comment"
          >
            <div
              class="sc-ksJisA hKuXOJ card"
            >
              <div
                class="sc-fnOeiS dGYhpH card-header"
              >
                <div
                  class="sc-jIJgYh gPIvtQ"
                >
                  <div
                    class="sc-jdkVqZ ivvINy"
                  >
                    <span
                      class="sc-hhWzdI sc-gXCJSa gPjntB fqfOft"
                      data-testid="filled"
                    >
                      ★
                    </span>
                  </div>
                  <div
                    class="sc-jdkVqZ ivvINy"
                  >
                    <span
                      class="sc-hhWzdI sc-gXCJSa gPjntB fqfOft"
                      data-testid="filled"
                    >
                      ★
                    </span>
                  </div>
                  <div
                    class="sc-jdkVqZ ivvINy"
                  >
                    <span
                      class="sc-hhWzdI sc-gXCJSa gPjntB fqfOft"
                      data-testid="filled"
                    >
                      ★
                    </span>
                  </div>
                  <div
                    class="sc-jdkVqZ ivvINy"
                  >
                    <span
                      class="sc-hhWzdI sc-gXCJSa gPjntB fqfOft"
                      data-testid="filled"
                    >
                      ★
                    </span>
                  </div>
                  <div
                    class="sc-jdkVqZ ivvINy"
                  >
                    <span
                      class="sc-hhWzdI sc-gXCJSa gPjntB fqfOft"
                      data-testid="filled"
                    >
                      ★
                    </span>
                  </div>
                  5
                   
                  /5
                </div>
              </div>
              <div
                class="sc-hBpgZr fzdmTd card-body"
              >
                <p
                  class="sc-fEyylQ bjhgma card-text"
                >
                  Great product!
                </p>
                <p
                  class="sc-bjEwCx bPzSwy card-text"
                >
                  author:
                  John Doe
                   
                  <br />
                   review id:
                  1
                </p>
              </div>
            </div>
          </div>
          <div
            data-testid="comment"
          >
            <div
              class="sc-ksJisA hKuXOJ card"
            >
              <div
                class="sc-fnOeiS dGYhpH card-header"
              >
                <div
                  class="sc-jIJgYh gPIvtQ"
                >
                  <div
                    class="sc-jdkVqZ ivvINy"
                  >
                    <span
                      class="sc-hhWzdI sc-gXCJSa gPjntB fqfOft"
                      data-testid="filled"
                    >
                      ★
                    </span>
                  </div>
                  <div
                    class="sc-jdkVqZ ivvINy"
                  >
                    <span
                      class="sc-hhWzdI sc-gXCJSa gPjntB fqfOft"
                      data-testid="filled"
                    >
                      ★
                    </span>
                  </div>
                  <div
                    class="sc-jdkVqZ ivvINy"
                  >
                    <span
                      class="sc-hhWzdI sc-gXCJSa gPjntB fqfOft"
                      data-testid="filled"
                    >
                      ★
                    </span>
                  </div>
                  <div
                    class="sc-jdkVqZ ivvINy"
                  >
                    <span
                      class="sc-hhWzdI gPjntB"
                      data-testid="empty"
                    >
                      ★
                    </span>
                  </div>
                  <div
                    class="sc-jdkVqZ ivvINy"
                  >
                    <span
                      class="sc-hhWzdI gPjntB"
                      data-testid="empty"
                    >
                      ★
                    </span>
                  </div>
                  3
                   
                  /5
                </div>
              </div>
              <div
                class="sc-hBpgZr fzdmTd card-body"
              >
                <p
                  class="sc-fEyylQ bjhgma card-text"
                >
                  Could be better.
                </p>
                <p
                  class="sc-bjEwCx bPzSwy card-text"
                >
                  author:
                  Jane Smith
                   
                  <br />
                   review id:
                  2
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
